src/util/std_types.h

```diff
@@ -151,6 +151,13 @@ inline symbol_typet &to_symbol_type(typet &type)
   return static_cast<symbol_typet &>(type);
 }
 
+template <>
+inline bool can_cast_type<symbol_typet>(const typet &type)
```

Comment:
Could this please be factored out into a separate commit and PR? It's the only reason I get invited to the reviewing party, and this bit looks good and useful to me. For all the rest I have no clue...

Reply:
Sorry I didn't think the faff of two branches, having to rebase the later on the former when one is merged etc was worth the hassle - feel free to disregard the review request 

FWIW I increasingly view review requests that are automatically generated from code ownership are spurious and I wait till someone explicitly requests me before looking.

Reply:
Heh - just for the record I now want this in another PR, so I have been hoisted by my own laziness.